Extension/src/LanguageServer/configurations.ts

```diff
@@ -483,12 +483,6 @@ export class CppProperties {
                     resolve();
                 }, () => {});
             } else {
-                const settings: CppSettings = new CppSettings(this.rootUri);
-                if (providerId) {
-                    settings.update("default.configurationProvider", providerId);
```

Comment:
I removed this arbitrarily add/erasing of settings.  It doesn't seem necessary, and reading/writing the user's settings seems like a terrible idea to me.  If they see it set a particular way, and we arbitrarily add/remove it out from under them, that's extremely confusing.